File: DailyCoding/Web-python/create.py
#!/usr/bin/python3
import html_sanitizer
import view
import os
import cgi
print("content-type:text/html; charset=UTF-8\n")
print("Hello World")

# 실패 로그 보는 법
# sudo tail /var/log/apache2/error.log

sanitizer = html_sanitizer.Sanitizer()

form = cgi.FieldStorage()

if 'id' in form:
    pageId = form["id"].value
    description = open('./data/'+pageId, 'r', encoding="utf-8").read()
    description = sanitizer.sanitize(description)

else:
    pageId = 'Welcome'
    description = 'Hello, web'


print('''<!doctype html>
<html>
<head>
  <title>WEB1 - Welcome</title>
  <meta charset="utf-8">
</head>
<body>
  <h1><a href="index.py">WEB</a></h1>
  <ol>
    {listStr}
  </ol>
  <a href="create.py">create</a>
  <form method="POST" action="./process_create.py">
    <p><input type="text" name="title" placeholder="title"></p>
    <p><textarea rows="10" name="description" placeholder="description"></textarea></p>
    <p><input type="submit"></p>
  </form>
</body>
</html>
'''.format(listStr=view.getList()))

File: DailyCoding/Web-PHP/create.php
<?php
require('print.php');
?>

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>
    <?php
    print_title();
    ?></title>
</head>
<body>
    <h1><a href="index.php">WEB</a></h1>
    <ol>
        <?php
            print_list();
        ?>
    </ol>
    <a href="create.php">create</a>
    <form action="create_process.php" method="post">
        <p><input type="text" name="title" placeholder="title"></p>
        <p><textarea name="description" id="" cols="30" rows="10" placeholder="Description">
        </textarea></p>
        <p><input type="submit"></p>
    </form>
    <h2>
        <?php
            print_title();
        ?>
    </h2>
    <?php
        print_description();
    ?>
</body>
</html>

